robot_skills/src/robot_skills/picovoice.py

```diff
@@ -0,0 +1,40 @@
+import rospy
+
+from hmi.client import TimeoutException
+from hmi.common import HMIResult
+from picovoice_msgs.msg import GetIntentAction, GetIntentGoal, GetIntentResult
+from robot_skills.robot_part import RobotPart
+
+
+class PicoVoice(RobotPart):
+    def __init__(self, robot_name: str, tf_buffer):
+        super().__init__(robot_name=robot_name, tf_buffer=tf_buffer)
+        self._action_client = self.create_simple_action_client(f"/{robot_name}/get_intent", GetIntentAction)
+
+    def get_intent(self, context_url: str, require_endpoint=True, timeout: float = 10.0) -> HMIResult:
+        """
+        get_intent
+
+        :param context_url:
+        :param require_endpoint:
+        :param timeout:
+        :return:
+        :raises TimeoutException: In case of timeout
+        """
+        timeout = rospy.Duration.from_sec(timeout)
+        goal = GetIntentGoal(context_url=context_url, require_endpoint=require_endpoint)
+        rospy.logdebug(f"PicoVoice.get_intent: {goal=}")
+        if not self._action_client.send_goal_and_wait(goal, execute_timeout=timeout, preempt_timeout=timeout):
+            rospy.logerr("PicoVoice.get_intent: failed")
+            raise TimeoutException("PicoVoice.get_intent: failed")
+
+        result = self._action_client.get_result()  # type: GetIntentResult
+        rospy.logdebug(f"PicoVoice.get_intent: {result=}")
+
+        if result is None:
+            rospy.logerr("PicoVoice.get_intent: result: None")
+            raise TimeoutException("PicoVoice.get_intent: result: None")
+        if not result.is_understood:
+            rospy.logwarn("PicoVoice.get_intent: Not understood")
+            raise TimeoutException("PicoVoice.get_intent: Not understood")
```

Comment:
These are not strictly timeoutexceptions. They also conflict with the docstring which states that it will only raise a timeoutexception in case of timeout